File: tutorial/codelab/move_zero/move_zero.cpp
#include <iostream>
using namespace std;
void moveZeroes(int arr[], int len)
{

    int start = 0;
    int end = len;

    for (int j = 0; j < len; j++)
    {
        if (arr[j] != 0)
        {
            swap(arr[j], arr[start]);
            start++;
        }
    }

    for (int i = 0; i < len; i++)
    {
        cout << arr[i] << " ";
    }
}
int main()
{
    int arr[8] = {0, 1, 0, 5, 0, 8, 7, 9};

    moveZeroes(arr, 8);
}

File: tutorial/codelab/move_zero/move_zero.dart
void main() {
  int start = 0;
  List<int> arr = [0, 1, 0, 5, 0, 8, 7, 9];

  for (var i = 0; i < arr.length; i++) {
    if (arr[i] != 0) {
      int a = arr[i];
      int b = arr[start];

      arr[i] = b;
      arr[start] = a;
      start++;
    }
  }

  print(arr);
}
